cml-tdengine/src/core/inference.rs

```diff
@@ -149,6 +170,204 @@ mod tests {
+            .data_path("/tmp/inference_dir/inference_data1.txt".into())
+            .build()?];
+
+        let available_status = vec!["SUCCESS"];
+        let last_batch_time_1: i64 = taos
+            .query_one(format!(
+                "SELECT LAST(ts) FROM task.`{}` WHERE status IN ({}) ",
+                batch_meta_1.batch(),
+                available_status
+                    .iter()
+                    .map(|s| format!("'{}'", s))
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            ))
+            .await?
+            .unwrap();
+        let last_batch_time_2: i64 = taos
+            .query_one(format!(
+                "SELECT LAST(ts) FROM task.`{}` WHERE status IN ({}) ",
+                batch_meta_2.batch(),
+                available_status
+                    .iter()
+                    .map(|s| format!("'{}'", s))
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            ))
+            .await?
+            .unwrap();
+        fs::write(
+            "/tmp/inference_dir/".to_string()
+                + batch_meta_1.batch()
+                + &last_batch_time_1.to_string()
+                + ".txt",
+            b"10",
+        )?;
+        fs::write(
+            "/tmp/inference_dir/".to_string()
+                + batch_meta_2.batch()
+                + &last_batch_time_2.to_string()
+                + ".txt",
+            b"20",
+        )?;
+        let inference_fn = |vec_data: &mut Vec<NewSample<Value>>,
+                            batch: &str,
+                            task_time: i64|
+         -> Vec<NewSample<Value>> {
+            let mut result: Vec<NewSample<Value>> = Vec::new();
+            let working_dir = "/tmp/inference_dir/".to_string();
+            let model_inference =
+                fs::read_to_string(working_dir + batch + &task_time.to_string() + ".txt")
+                    .unwrap()
+                    .parse::<f32>()
+                    .unwrap();
+            // inference
+            let inference_data_fn = |inference_data, inference_model| {
+                if inference_data + inference_model > 25.0 {
+                    Some(Value::Float(inference_data + model_inference))
+                } else {
+                    None
+                }
+            };
+            for inference_data in vec_data.iter() {
+                let output = inference_data_fn(
+                    fs::read_to_string(inference_data.data_path())
+                        .unwrap()
+                        .parse::<f32>()
+                        .unwrap(),
+                    model_inference,
+                );
+                result.push(
+                    NewSampleBuilder::default()
+                        .data_path(inference_data.data_path().to_path_buf())
+                        .output(output)
+                        .build()
+                        .unwrap(),
+                );
+            }
```

Comment:
There's no need for use a closure here, for the closure won't be used more than one time. Just move the calculations into loop, and use variables for staging intermediate results if necessary.